style: move ALWAYS_INLINE to the beginning to resolve compiler warnings

laroche opened the pull request:
To resolve the following gcc compiler warning:
zephyr/arch/arm/core/aarch32/cortex_a_r/fault.c:101:1: warning: 'inline' is not at beginning of declaration [-Wold-style-declaration]
  101 | static void ALWAYS_INLINE z_arm_fpu_caller_save(struct __fpu_sf *fpu)
      | ^~~~~~

I searched to all of the source code to find these further occurances where inline is not at the beginning of a function declaration.

dcpleung commented:
Which compiler complains about this?

laroche commented:
This is gcc -Wold-style-declaration from the zephyr sdk.

best regards,

Florian La Roche

povergoing commented:
> This is gcc -Wold-style-declaration from the zephyr SDK.

You need to fix your commit message for exceeding 75 chars, and it might be better to append your answer to the commit message BTW.

laroche committed:
style: move ALWAYS_INLINE to the beginning to resolve compiler warnings

With gcc from the zephyr sdk and -Wold-style-declaration is giving this
output:
zephyr/arch/arm/core/aarch32/cortex_a_r/fault.c:101:1: warning:
  'inline' is not at beginning of declaration [-Wold-style-declaration]
  101 | static void ALWAYS_INLINE
                    z_arm_fpu_caller_save(struct __fpu_sf *fpu)
      | ^~~~~~

I searched to all of the source code to find these further occurances
where inline is not at the beginning of a function declaration.

Signed-off-by: Florian La Roche <Florian.LaRoche@gmail.com>
